src/core/server/rendering/rendering_service.test.ts

```diff
@@ -0,0 +1,185 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { load } from 'cheerio';
+
+import { httpServerMock } from '../http/http_server.mocks';
+import { uiSettingsServiceMock } from '../ui_settings/ui_settings_service.mock';
+import { mockRenderingServiceParams, mockRenderingSetupDeps } from './__mocks__/params';
+import { RenderingServiceSetup } from './types';
+import { RenderingService } from './rendering_service';
+
+const INJECTED_METADATA = {
+  version: expect.any(String),
+  branch: expect.any(String),
+  buildNumber: expect.any(Number),
+  env: {
+    binDir: expect.any(String),
+    configDir: expect.any(String),
+    homeDir: expect.any(String),
+    logDir: expect.any(String),
+    packageInfo: {
+      branch: expect.any(String),
+      buildNum: expect.any(Number),
+      buildSha: expect.any(String),
+      version: expect.any(String),
+    },
+    pluginSearchPaths: expect.any(Array),
+    staticFilesDir: expect.any(String),
+  },
+  legacyMetadata: {
+    branch: expect.any(String),
+    buildNum: expect.any(Number),
+    buildSha: expect.any(String),
+    version: expect.any(String),
+  },
+};
+const { createKibanaRequest, createRawRequest } = httpServerMock;
+const legacyApp = { getId: () => 'legacy' };
+
+describe('RenderingService', () => {
+  let service: RenderingService;
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    service = new RenderingService(mockRenderingServiceParams);
+  });
+
+  describe('setup()', () => {
+    it('creates instance of RenderingServiceSetup', async () => {
+      const rendering = await service.setup(mockRenderingSetupDeps);
+
+      expect(rendering.render).toBeInstanceOf(Function);
+    });
+
+    describe('render()', () => {
```

Comment:
don't see **explicit** tests for:
- uiPlugins
- uiPublicUrl
- bootstrapScriptUrl
- legacyMode
- locale
etc.

Reply:
https://elastic.slack.com/archives/C5TQ33ND8/p1576613947075400